### ISSUE TITLE
Cheat codes

### ISSUE THREAD

**Issue by Retartios** [calc84maniac/tiboyce#146]
Are you guys going to add the ability to add cheat codes to the emulator? 
I'd like to have more fun with the current games I play but getting the items is too time consuming.
There are already existing cheat codes on game shark and other emulators, such as GTA and Pokemon.
Is there any hope for cheat codes coming to this emulator?

**Comment by calc84maniac** [comment]
Yeah, that's something I could potentially add, at least for GameShark codes. Game Genie codes might be a problem because they temporarily modify the ROM itself, which is difficult because TI-Boy doesn't load ROMs into RAM. Though, it may be possible to do it with a limited number of codes active at once, based on available RAM to load and modify portions of the ROM.

**Comment by Icey-Glitch** [comment]
any status on this?

**Comment by calc84maniac** [comment]
Not yet, but the ROM mapping changes for the GBC release should help make limited Game Genie support easier. This might be a good feature to prioritize for the next minor releases.

**Comment by Gaphilly-real** [comment]
Sorry, any progress yet?

**Comment by calc84maniac** [comment]
It's the next thing on my to-do list, actually. I've been a bit distracted in the meantime by [CEmu](https://github.com/CE-Programming/CEmu) development (which will make TI-Boy CE fully usable in that emulator), but in the meantime I've made some progress towards supporting on-the-fly ROM patching during emulation.

**Comment by Gaphilly-real** [comment]
Ok cool!

**Comment by sig764468-commits** [comment]
has there been any update on this yet

**Comment by RichieGaming27** [comment]
> Yeah, that's something I could potentially add, at least for GameShark codes. Game Genie codes might be a problem because they temporarily modify the ROM itself, which is difficult because TI-Boy doesn't load ROMs into RAM. Though, it may be possible to do it with a limited number of codes active at once, based on available RAM to load and modify portions of the ROM.

It's already may 2026, is there any progress? GameShark codes is already enough. Please can you do smt on it?
